### ISSUE TITLE
Generator is trash

### ISSUE THREAD

**Issue by aaryswastaken** [aaryswastaken/gandi-rush#39]
works ok-tier when gems >= 5

**Comment by aaryswastaken** [comment]
I know how to treat this problem. Works with n_gem >= 4.

~~It's not the best algo because ideal albo will be able to make reasonable entropy with n_gem >= 3 but still.~~
*EDIT: See at the conclusion. Might be optimised for n_gem >= 3 as well.*

The idea is that given a certain grid, we generate half of the cell purely randomly like this:

```md
    x 1 x 2 x 0 x 3 
    1 x 0 x 3 x 1 x
    x 2 x 1 x 2 x 0
    3 x 1 x 3 x 0 x
```

Then the idea is to generate a second grid that is the "used grid". For now the grid is all 0:

```
    0 0 0 0 0 0 0 0 
    0 0 0 0 0 0 0 0
    0 0 0 0 0 0 0 0 
    0 0 0 0 0 0 0 0
```

Let's take the cell in (x, y) = (5, 1):

```md
    . 0 .         . 0 .
    3 x 1    /    0 0 0 
    . 2 .         . 0 . 
```

We will generate a random number in the interested range, here we will chose 2:

The we will update the usage grid.

```md
    . 0 .         . 0 .
    3 2 1    /    0 1 0
    . 2 .         . 1 .
```

Notice that the 2 at the bottom (5, 2) has turned into a 1 in the usage matrix because it is adjacent to another cell with the same number 

Now we will see the case of (6, 2):

```md
    . 1 .       . 0 .
    2 x 0   /   1 0 0 
    . 0 .       . 0 . 
```

When generating a random number, we have once again a 2. Now let's look at this array: `[2]`.
This array has been generated by filtering every 4 adjacent cell's values by if they are used or not.

Because the 2 is already in the array, we are going to use another number: 0. As 0 is not in the extracted array,  we know this is a valid choice.

Let's update both matrix. Here is the extract:

```md
    . 1 .       . 0 .
    2 0 0   /   1 1 1
    . 0 .       . 1 .
```

Note: here we chose the number then compare it to the adjacence array. In reality we take the opposite array (opposite of `[2]` would be `[0, 1, 3]`), then taking a random number into this array by generating a random index. This dodges a loop of test.

This algorithm is robust because in the example we covered, we chose random cells but in the real implementation, in reality we start from a corner then start to develop everything from this. Thus, in the worse case scenario, only two items are in the adjacence array, thus letting always an element (for n_gem = 3) or more.

Note: I need to think a bit more to assure that it works robustly for n_gem = 3.
